File: Python/4-Array-based-Programs/53-find-duplicate-element-in-array-N+1.py
def Duplicate(arr):
    single= arr[0]
    double = arr[0]

    while True:
        single= arr[single]
        double=arr[arr[double]]

        if single == double:
            break
    
    single = arr[0]

    while single != double:
        single = arr[single]
        double = arr[double]

    return single 


arr = [1, 5, 3, 2, 5, 4]
print("The Duplicate element is:", Duplicate(arr))

File: Java/4-Array-based-Programs/53-find-duplicate-element-in-array-N+1.java
class Main {
    static int Duplicate(int[] arr){
        int Single = arr[0];
        int Double = arr[0];

        while (true){
            Single=arr[Single];
            Double=arr[arr[Double]];

            if (Single == Double){
                break;
            }
        }

        Single = arr[0];
        while (Single != Double){
            Single = arr[Single];
            Double = arr[Double];
        }

        return Single;
    }
    
    public static void main(String[] args) {
        int[] arr = {2, 1, 5, 3, 4, 3};
        System.out.println("The Duplicate value is: " + Duplicate(arr));
    }
}